tools/base/fpgad/command_line.c

```diff
@@ -294,6 +312,57 @@ void cmd_canonicalize_paths(struct fpgad_config *c)
 				"%s/%s", c->directory, tmp);
 	}
 	LOG("daemon pid file is %s\n", c->pidfile);
+
+	// Verify cfgfile doesn't contain ".."
+	def = false;
+	sub = NULL;
+	strstr_s(c->cfgfile, sizeof(c->cfgfile),
+			"..", 2, &sub);
+	if (sub)
+		def = true;
+
+	if (def || (c->cfgfile[0] == '\0')) {
+		search = true;
+	} else {
+		char *canon_path;
+
+		canon_path = canonicalize_file_name(c->cfgfile);
+		if (canon_path) {
+			if (!stat(canon_path, &stat_buf)) {
```

Comment:
I think `canonicalize_file_name`  returns NULL if a component of the path doesn't exist.
http://man7.org/linux/man-pages/man3/canonicalize_file_name.3.html
![image](https://user-images.githubusercontent.com/22808196/52376715-a9080400-2a17-11e9-88c0-12838cd28fb4.png)

Reply:
Ok, will remove the stat.